### PR TITLE
BUG: fixed reference leak when SWIG passing tuples or lists

### LINKED ISSUE #2705: Unlimited memory usage when calling several `Image` and `Region` methods from Python
### Description
The below Python programs seem to require unlimited memory.

### Steps to Reproduce
This is a family of seemingly related issues, which have roughly the same setup. Run the below code to set up a few examples

```python
import itk

Dimension = 2
PixelType = itk.F
ImageType = itk.Image[PixelType, Dimension]

image = ImageType.New()

size = itk.Size[Dimension]()
size.Fill(500)

index = itk.Index[Dimension]()
index.Fill(50)

RegionType = itk.ImageRegion[Dimension]
region = RegionType()
region.SetIndex(index)
region.SetSize(size)

image.SetRegions(region)
image.Allocate()
```

With the above setup, run one of the following code cells and monitor memory usage. They all consume an unlimited amount of memory:

1.) `Region::IsInside` with tuple argument with x or y >= 256 or <= -6
```python
x, y = 256, 255
for i in range(10000000000000):
    region.IsInside((x + (i % 2), y + (i % 2)))
```

2.) `Image::TransformPhysicalPointToIndex` with tuple argument with x or y >= 256  or <= -6
```python
x, y = 256, 255
for i in range(10000000000):
    image.TransformPhysicalPointToIndex((x + (i % 2), y + (i % 2)))
```

3.) `Image::TransformPhysicalPointToIndex` with `itk.Point` argument with ANY value of x or y (as far as I can tell)
```python
x, y = 0, 0
for i in range(10000000000):
    p = itk.Point[PixelType, Dimension]()
    p[0] = x + (i % 2)
    p[1] = y + (i % 2)
    image.TransformPhysicalPointToIndex(p)
```

4.) `Image::TransformIndexToPhysicalPoint` with tuple argument with x or y >= 256 or <= -6
```python
x, y = 256, 255
for i in range(10000000000):
    image.TransformIndexToPhysicalPoint((x + (i % 2), y + (i % 2)))
```

Swapping tuples for `itk.Index` arguments for cases 1 and 4 resolves the issue. From [this](https://www.codementor.io/@arpitbhayani/how-python-implements-super-long-integers-12icwon5vk) source:

"Python preallocates small integers in a range of -5 to 256. This allocation happens during initialization and since we cannot update integers (immutability) these preallocated integers are singletons and are directly referenced instead of reallocating. This means every time we use/creates a small integer, python instead of reallocating just returns the reference of preallocated one.". Thanks @brad-t-moore 

This gives a hint to the above behavior.

### Actual behavior

```none
```
### Versions

ITK 5.2.0
### Environment
Python 3.6.9, Ubuntu 18.04

### Additional Information

Valgrind output for case 1:

```
==27314== LEAK SUMMARY:
==27314==    definitely lost: 64 bytes in 1 blocks
==27314==    indirectly lost: 0 bytes in 0 blocks
==27314==      possibly lost: 306,216 bytes in 200 blocks
==27314==    still reachable: 31,665,705 bytes in 36,532 blocks
==27314==                       of which reachable via heuristic:
==27314==                         stdstring          : 445,212 bytes in 5,969 blocks
==27314==         suppressed: 0 bytes in 0 blocks
```

Valgrind gives the same summary when swapping a tuple out for an `itk.Index` argument in case 1

### PR DESCRIPTION
**Pull request by brad-t-moore**
Fixes #2705.  Calls to PySequence_GetItem have been updated in pyBase.i to decrement the reference count correctly.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [X] Updated API documentation (or API not changed)
- [X] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [X] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [X] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by brad-t-moore** [comment]
I didn't add a test for because I never did figure out a reliable way to get at the unreferenced PyObjects.  I have confirmed that the memory overflows in local build of ITK without this change and does not afterwards.

File: Wrapping/Generators/Python/PyBase/pyBase.i [hunk 1]
```diff
@@ -603,9 +603,11 @@ str = str
                     } else if (PyFloat_Check(o)) {
                         itks[i] = (type)PyFloat_AsDouble(o);
                     } else {
+						Py_DECREF(o);
```
Comment (dzenanz) [inline Wrapping/Generators/Python/PyBase/pyBase.i:606]: Code alignment looks very wrong.

**Review by dzenanz — COMMENTED** [review]
I can't really judge the substance of this patch, but I trust you 😄

**Review by dzenanz — APPROVED** [review]

**Review by thewtex — APPROVED** [review]
:1st_place_medal: awesome @brad-t-moore !

**Comment by brad-t-moore** [comment]
Thanks @thewtex  !  It was a team effort with @tao558 !  He did a phenomenal job diligently tracking this reference leak all the way to the SWIG components.

**Pull request merged**